gcloud-java-compute/src/test/java/com/google/gcloud/compute/OperationTest.java

```diff
@@ -0,0 +1,489 @@
+        .status(STATUS)
+        .statusMessage(STATUS_MESSAGE)
+        .user(USER)
+        .progress(PROGRESS)
+        .insertTime(INSERT_TIME)
+        .startTime(START_TIME)
+        .endTime(END_TIME)
+        .errors(ERRORS)
+        .warnings(WARNINGS)
+        .httpErrorStatusCode(HTTP_ERROR_STATUS_CODE)
+        .httpErrorMessage(HTTP_ERROR_MESSAGE)
+        .description(DESCRIPTION)
+        .build();
+    zoneOperation = new Operation.Builder(serviceMockReturnsOptions)
+        .id(ID)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .operationId(ZONE_OPERATION_ID)
+        .clientOperationId(CLIENT_OPERATION_ID)
+        .operationType(OPERATION_TYPE)
+        .targetLink(TARGET_LINK)
+        .targetId(TARGET_ID)
+        .status(STATUS)
+        .statusMessage(STATUS_MESSAGE)
+        .user(USER)
+        .progress(PROGRESS)
+        .insertTime(INSERT_TIME)
+        .startTime(START_TIME)
+        .endTime(END_TIME)
+        .errors(ERRORS)
+        .warnings(WARNINGS)
+        .httpErrorStatusCode(HTTP_ERROR_STATUS_CODE)
+        .httpErrorMessage(HTTP_ERROR_MESSAGE)
+        .description(DESCRIPTION)
+        .build();
+    regionOperation = new Operation.Builder(serviceMockReturnsOptions)
+        .id(ID)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .operationId(REGION_OPERATION_ID)
+        .clientOperationId(CLIENT_OPERATION_ID)
+        .operationType(OPERATION_TYPE)
+        .targetLink(TARGET_LINK)
+        .targetId(TARGET_ID)
+        .status(STATUS)
+        .statusMessage(STATUS_MESSAGE)
+        .user(USER)
+        .progress(PROGRESS)
+        .insertTime(INSERT_TIME)
+        .startTime(START_TIME)
+        .endTime(END_TIME)
+        .errors(ERRORS)
+        .warnings(WARNINGS)
+        .httpErrorStatusCode(HTTP_ERROR_STATUS_CODE)
+        .httpErrorMessage(HTTP_ERROR_MESSAGE)
+        .description(DESCRIPTION)
+        .build();
+    compute = createStrictMock(Compute.class);
+  }
+
+  private void initializeOperation() {
+    operation = new Operation.Builder(compute)
+        .id(ID)
+        .operationId(GLOBAL_OPERATION_ID)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .clientOperationId(CLIENT_OPERATION_ID)
+        .operationType(OPERATION_TYPE)
+        .targetLink(TARGET_LINK)
+        .targetId(TARGET_ID)
+        .status(STATUS)
+        .statusMessage(STATUS_MESSAGE)
+        .user(USER)
+        .progress(PROGRESS)
+        .insertTime(INSERT_TIME)
+        .startTime(START_TIME)
+        .endTime(END_TIME)
+        .errors(ERRORS)
+        .warnings(WARNINGS)
+        .httpErrorStatusCode(HTTP_ERROR_STATUS_CODE)
+        .httpErrorMessage(HTTP_ERROR_MESSAGE)
+        .description(DESCRIPTION)
+        .build();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    verify(serviceMockReturnsOptions);
+  }
+
+  @Test
+  public void testBuilder() {
+    initializeExpectedOperation(6);
+    assertEquals(CREATION_TIMESTAMP, globalOperation.creationTimestamp());
```

Comment:
I wonder if it's worth factoring any assertEquals common to all three operations into another function to reduce the size of this function.